lib/wab/impl/init.rb

```diff
@@ -33,6 +33,9 @@ def initialize(path, config)
 
         @verbose = config[:verbosity]
         @verbose = 'INFO' == @verbose || 'DEBUG' == @verbose || Logger::INFO == @verbose || Logger::DEBUG == @verbose
+        @write_cnt = 0
+        @exist_cnt = 0
+        @base_len = path.length + 1
```

Comment:
can you please explain the logic here..? `@base_len = path.length + 1`

Reply:
@base_len is used later to strip off the base from the full path of the files being copied. The extra 1 if for the trailing /.

Reply:
yep.. It clicked a few seconds after I posted the comment..

Reply:
I'm not sure about the performance, but instead of tracking the length of `"foo/" and then splicing our absolute paths, why not just write a helper function instead..?
```ruby
private
def relative_path(abs_path)
  abs_path.sub(@base_path, '')
end
```

Reply:
I'm pretty sure the way I have it is more efficient but for this it doesn't matter at all. To me it was the most obvious way to strip off the base. A helper would be fine but if write it I would still strip off the base with the length. Has more to do with our mental models of what a string is I think.